src/iop/watermark.c

```diff
@@ -400,6 +416,32 @@ int legacy_params(dt_iop_module_t *self,
     *new_version = 6;
     return 0;
   }
+  else if(old_version == 6)
+  {
+    const dt_iop_watermark_params_v6_t *o = (dt_iop_watermark_params_v6_t *)old_params;
+    dt_iop_watermark_params_v7_t *n = malloc(sizeof(dt_iop_watermark_params_v7_t));
+
+    n->opacity = o->opacity;
+    n->scale = o->scale;
+    n->xoffset = o->xoffset;
+    n->yoffset = o->yoffset;
+    n->alignment = o->alignment;
+    n->rotate = o->rotate;
+    n->scale_base = o->scale_base;
+    // let scale_img and scale_svg at the default values
```

Comment:
Those fields should also be copied, since they existed in the previous version....  As is, they become uninitialized data, since the memory is allocated with malloc rather than calloc.